### PR TITLE
conformance: Optimize mesh weight conformance tests using batch requests

### LINKED ISSUE #4101: v1.4.0: new conformance tests are too slow
The v1.4.0 conformance tests adds new 'Weight' tests. These are very very slow on a fast cluster:

```
--- PASS: TestGatewayConformance (38.44s)
    --- PASS: TestGatewayConformance/MeshGRPCRouteWeight (18.11s)
        --- PASS: TestGatewayConformance/MeshGRPCRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (18.10s)
    --- PASS: TestGatewayConformance/MeshHTTPRouteWeight (19.35s)
        --- PASS: TestGatewayConformance/MeshHTTPRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (19.35s)

```

In v1.3 the entire suite took 135.81s for us. So these 2 new tests add 27% additional time.

There is no need for this test to take so long. 

Running 500 simulations of the test logic: `echo {0..500} | xargs -P 10 -n1 kubectl exec echo-v1-67d4b56bd4-vdjc2 -- client grpc://echo:7070 --qps` takes 7s. Note the test intentionally adds 'sleeps' for some reason to the calls, hence why its 17s I imagine.

Doing 500 calls without port-forward takes 1s.

Using 1 call with `--count 500` takes 0.037s.

I would highly recommend we switch to `--count`

### PR DESCRIPTION
**Pull request by ciarams87**
**What this PR does / why we need it**:

The mesh weight conformance tests were executing 500 separate kubectl exec commands with random delays, resulting in very slow test execution.

This PR implements an optimization using the echo client's --count flag to execute all 500 requests in a single batch, dramatically reducing test time.

The new implementation maintains the same test behavior (validating
traffic distribution matches configured weights) while eliminating
the overhead of 500 separate kubectl exec calls and removing the
random delays that were adding significant wait time.

Testing:
Ran the mesh tests on istio in a GKE cluster

Performance improvement varies by environment - my tests showed the time went from 60-70s -> 3-4s per test (~95% faster)

Current main:
```
--- PASS: TestConformance (201.69s)
    --- PASS: TestConformance/MeshGRPCRouteWeight (64.10s)
        --- PASS: TestConformance/MeshGRPCRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (63.72s)
    --- PASS: TestConformance/MeshHTTPRouteWeight (66.91s)
        --- PASS: TestConformance/MeshHTTPRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (66.54s)
PASS
```

With the batch requests:
```
--- PASS: TestConformance (91.34s)
    --- PASS: TestConformance/MeshGRPCRouteWeight (3.51s)
        --- PASS: TestConformance/MeshGRPCRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (3.08s)
    --- PASS: TestConformance/MeshHTTPRouteWeight (3.92s)
        --- PASS: TestConformance/MeshHTTPRouteWeight/Requests_should_have_a_distribution_that_matches_the_weight (3.45s)
PASS
```

**Which issue(s) this PR fixes**:

Fixes #4101 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/utils/echo/pod.go [hunk 1]
```diff
@@ -275,3 +282,23 @@ func (m *MeshPod) CaptureRequestResponseAndCompare(t *testing.T, exp http.Expect
...
+	if err != nil {
+		return nil, fmt.Errorf("batch request failed: %w", err)
+	}
+
+	// Split the output by response boundaries
+	// Each response is separated by a blank line in the output
+	responses := parseMultipleResponses(resp.RawContent)
+
+	if len(responses) != count {
+		tlog.Logf(t, "Warning: expected %d responses but got %d", count, len(responses))
```
Comment (rikatz) [inline conformance/utils/echo/pod.go]: can you clarify why this is not an error? if you send 500 requests and gets 499 back, shouldn't this be a problem?

Comment (rikatz) [reply conformance/utils/echo/pod.go]: If this is expected, as I can see you have a "tolerance" configuration below, maybe the tolerance should be part of the MeshPod instance (so each test have a different tolerance) or please add a comment here raising that the caller of RequestBatch expects a tolerance.

Comment (ciarams87) [reply conformance/utils/echo/pod.go]: Good catch, thank you! Changed it to return an error. I made it a warning initially due to uncertainty about parsing edge cases, but you're right that missing responses should fail the test immediately. The tolerance is only for validating the weight distribution (±5%), not for handling missing responses.

Comment (rikatz) [reply conformance/utils/echo/pod.go]: thanks! I will come back to this review soon this week!

**Comment by LiorLieberman** [comment]
Hey @ciarams87, had a change to look at it?

**Comment by LiorLieberman** [comment]
> Thanks @ciarams87 and sorry for the delay, I did not get a notification for some reason.
> 
> Looks like you removed the random delay (`AddEntropy` function) why is that?
> 
> ref for why we added this: [#3827 (comment)](https://github.com/kubernetes-sigs/gateway-api/pull/3827#discussion_r2123121430)

Since we are not executing multiple commands, and instead using the count flag, this may actually be not necessary, since its just one command execution. 

So looks good overall. 

The only thing is that I ran it myself now, and I dont see anything that confirms that 500 requests were actually executed.

**Comment by LiorLieberman** [comment]
~ok just merged https://github.com/istio/istio/pull/58454 to fix it~

The above proved to be unnecessary 😅

File: conformance/utils/echo/pod.go [hunk 2]
```diff
@@ -275,3 +282,23 @@ func (m *MeshPod) CaptureRequestResponseAndCompare(t *testing.T, exp http.Expect
...
+
+	resp, err := m.request(req)
+	if err != nil {
+		return nil, fmt.Errorf("batch request failed: %w", err)
+	}
+
+	// Split the output by response boundaries
+	// Each response is separated by a blank line in the output
+	responses := parseMultipleResponses(resp.RawContent)
+
```
Comment (LiorLieberman) [inline conformance/utils/echo/pod.go:298]:
```suggestion
	responses := parseMultipleResponses(resp.RawContent)

    tlog.Logf(t, "Processed %d requests", len(responses))

```

**Comment by LiorLieberman** [comment]
Added one nit. 

@ciarams87 let me know if you would like to finalize this or I can open a PR. This one is an important fix

**Comment by LiorLieberman** [comment]
Feedback from today's community meeting: Since we want to move forward to include this in the next patch release its better if we merged it soon. I am going to merge the PR I opened (which relies on the same commits). Thanks @ciarams87 for driving this again.

**Comment by kflynn** [comment]
Closing in favor of #4300. Thanks @ciarams87!! 🙂
